lib/lookup-table/include/LookupTable.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include <iterator>
+#include <optional>
+#include <unordered_map>
+
+// This a LUT type. It is not meant to be mutated after initilization.
+template <class Key, class Value, class Hash = std::hash<Key>, class KeyEqual = std::equal_to<Key>,
+          class Allocator = std::allocator<std::pair<const Key, Value>>>
+class LookupTable {
+ private:
+  const std::unordered_map<Key, Value, Hash, KeyEqual, Allocator> _unordered_map;
+
+ public:
+  using key_type        = Key;
+  using mapped_type     = Value;
+  using value_type      = std::pair<const Key, Value>;
+  using size_type       = size_t;
+  using difference_type = std::ptrdiff_t;
+  using hasher          = Hash;
+  using key_equal       = KeyEqual;
+  using allocator_type  = Allocator;
+  using reference       = value_type &;
+  using const_reference = const value_type &;
+  using pointer         = std::allocator_traits<Allocator>::pointer;
+  using const_pointer   = std::allocator_traits<Allocator>::const_pointer;
+
+  LookupTable() = delete;
+
+  // range based constructor
+  template <typename InputIterator>
+  LookupTable(InputIterator first, InputIterator last) : _unordered_map(first, last) {}
+  // copy constructor and assignment
+  LookupTable(const LookupTable &copy) = default;
+  LookupTable &operator=(const LookupTable &) = default;
+  // move constructor and assignment
+  LookupTable(LookupTable &&move) = default;
+  LookupTable &operator=(LookupTable &&) = default;
+  // initializer list constructor
+  LookupTable(std::initializer_list<value_type> init, size_type bucket_count = 0, const Hash &hash = Hash(),
+              const key_equal &equal = key_equal(), const Allocator &alloc = Allocator())
+      : _unordered_map(init, bucket_count, hash, equal, alloc) {}
```

Comment:
What if there was an option or param that was like the default return value? cause like in most situations, id imagine that people would be like LUT[key].value_or(default_value). what if you chose to provide the LUT with a default value, it could just return the default instead of having value_or all over the place?

Reply:
yeah i can do that. itd just be an optional argument

Reply:
also when i mean provide, i mean like to constructor, not to function call